File: api_service/models/post.py
from datetime import datetime
from typing import Optional

import orjson
from pydantic import BaseModel

from models.common import orjson_dumps
from models.forum import Forum
from models.thread import Thread
from models.user import User


class Post(BaseModel):
    id: int = 0
    message: str
    created: datetime = None
    author: str
    forum: str = ''
    isEdited: bool = False
    parent: Optional[int] = None
    thread: int = 0

    class Config:
        # Заменяем стандартную работу с json на более быструю
        json_loads = orjson.loads
        json_dumps = orjson_dumps


class FullPost(BaseModel):
    post: Post
    author: User = None
    thread: Thread = None
    forum: Forum = None

    class Config:
        # Заменяем стандартную работу с json на более быструю
        json_loads = orjson.loads
        json_dumps = orjson_dumps


class PostUpdate(BaseModel):
    message: str = ''

    class Config:
        # Заменяем стандартную работу с json на более быструю
        json_loads = orjson.loads
        json_dumps = orjson_dumps

File: api_service/models/post.go
package models

import "time"

//easyjson:json
type Post struct {
	Id       int       `json:"id,omitempty"`
	Message  string    `json:"message,omitempty"`
	Created  time.Time `json:"created,omitempty"`
	Author   string    `json:"author,omitempty"`
	Forum    string    `json:"forum,omitempty"`
	IsEdited bool      `json:"isEdited,omitempty"`
	Parent   int       `json:"parent,omitempty"`
	Thread   int       `json:"thread,omitempty"`
}

//easyjson:json
type PostUpdate struct {
	Message string `json:"message,omitempty"`
}

//easyjson:json
type FullPost struct {
	Post   Post   `json:"post,omitempty"`
	Author User   `json:"author,omitempty"`
	Thread Thread `json:"thread,omitempty"`
	Forum  Forum  `json:"forum,omitempty"`
}
